src/auth/auth.service.ts

```diff
@@ -25,10 +26,21 @@ export const AuthService = new (class {
                 first_name: data.first_name,
                 last_name: data.last_name,
                 birthdate: data.birthdate,
-                passwordHash: this.hashPassword(data.password)
+                passwordHash: this.hashPassword(data.password),
+                status: UserStatus.BLOCKED
```

Comment:
You need to add another user status `UNVERIFIED` or another column `verified: bool`.
BLOCKED does not equal UNVERIFIED.
And we should rename BLOCKED to BANNED because it will be clearer what it really does.